src/main/kotlin/no/nav/tiltaksarrangor/consumer/DeltakerlisteHandler.kt

```diff
@@ -0,0 +1,93 @@
+package no.nav.tiltaksarrangor.consumer
+
+import com.fasterxml.jackson.module.kotlin.readValue
+import no.nav.amt.lib.utils.objectMapper
+import no.nav.tiltaksarrangor.client.amtarrangor.AmtArrangorClient
+import no.nav.tiltaksarrangor.client.amtarrangor.dto.toArrangorDbo
+import no.nav.tiltaksarrangor.consumer.KafkaConsumer.Companion.DELTAKERLISTE_V2_TOPIC
+import no.nav.tiltaksarrangor.consumer.model.DeltakerlistePayload
+import no.nav.tiltaksarrangor.repositories.ArrangorRepository
+import no.nav.tiltaksarrangor.repositories.DeltakerlisteRepository
+import no.nav.tiltaksarrangor.repositories.TiltakstypeRepository
+import no.nav.tiltaksarrangor.unleash.UnleashToggle
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.util.UUID
+
+@Service
+class DeltakerlisteHandler(
+	private val arrangorRepository: ArrangorRepository,
+	private val deltakerlisteRepository: DeltakerlisteRepository,
+	private val tiltakstypeRepository: TiltakstypeRepository,
+	private val amtArrangorClient: AmtArrangorClient,
+	private val unleashToggle: UnleashToggle,
+) {
+	private val log = LoggerFactory.getLogger(javaClass)
+
+	fun lagreDeltakerliste(
+		topic: String,
+		deltakerlisteId: UUID,
+		value: String?,
+	) {
+		if (value == null) {
+			deltakerlisteRepository.deleteDeltakerlisteOgDeltakere(deltakerlisteId)
+			log.info("Slettet tombstonet deltakerliste med id $deltakerlisteId")
+			return
+		}
+
+		if (topic == DELTAKERLISTE_V2_TOPIC && !unleashToggle.skalLeseGjennomforingerV2()) {
+			log.info("Unleash er ikke enabled for $DELTAKERLISTE_V2_TOPIC")
+			return
+		}
+
+		val tiltakskodeFromJson = getTiltakskodeFromMessageJson(value)
+		if (!unleashToggle.erKometMasterForTiltakstype(tiltakskodeFromJson)) {
+			log.info("Tiltakskode $tiltakskodeFromJson er ikke støttet.")
+			return
+		}
+
+		val deltakerlistePayload: DeltakerlistePayload = objectMapper.readValue(value)
+
+		if (deltakerlistePayload.skalLagres()) {
```

Comment:
Dette blir vel ikke helt ekvivalent med det som var. Det var jo en filtrering på tiltakstype her som ikke har med om komet er master eller ikke. Vi blir jo senere master for de nye enkeltplassene men de skal ikke dukke opp her

Reply:
Endret koden til
```
		val tiltakskodeFromJson =
			getTiltakskodeFromDeltakerlisteJson(value)
				?: getSecondaryTiltakskodeFromDeltakerlisteJson(value)
		if (!tiltakskodeErStottet(tiltakskodeFromJson)) {
			log.info("Tiltakskode $tiltakskodeFromJson er ikke støttet.")
			return
		}
```

slik at prosesseringen stopper så tidlig som mulig.